src/mixins/helpers.js

```diff
@@ -190,17 +190,27 @@ var helpers = {
       targetLeft = currentLeft;
     }
 
-    if(this.props.afterChange!==null){
-      this.props.afterChange(currentSlide);
+    if (this.props.beforeChange) {
+      this.props.beforeChange(currentSlide);
     }
 
-    var callback = function() {
-        this.setState({
-          animating: false,
-          trackStyle: this.getCSS(currentLeft),
-          swipeLeft: null
-        });
-    }.bind(this);
+    var nextStateChanges = {
+      animating: false,
+      trackStyle: this.getCSS(currentLeft),
+      swipeLeft: null
+    };
+
+    var callback1 = () => {
+      this.setState(nextStateChanges);
+    };
+
+    var callback2 = () => {
+      this.setState(nextStateChanges);
+
+      if (this.props.afterChange) {
+        this.props.afterChange(currentSlide);
+      }
+    };
```

Comment:
@akiran Instead of callback1 and callback2. Can this be written more nicely with a single callback?